src/components/v2/Button/FormikSubmitButton.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { useFormikContext } from 'formik';
+import { PrimaryButton, IButtonProps } from '.';
+
+interface IFormikSubmitButtonProps extends IButtonProps {
+  disabled?: boolean;
+  loading: boolean;
+  disabledLabel?: string;
+  enabledLabel: string;
+}
+
+export const FormikSubmitButton = ({
+  disabledLabel,
+  enabledLabel,
+  variant = 'primary',
+  disabled = false,
+  loading = false,
+  ...rest
+}: IFormikSubmitButtonProps) => {
+  const { isValid, dirty } = useFormikContext();
+  const disable = !isValid || !dirty;
+  return (
+    <PrimaryButton
+      variant={variant}
+      {...rest}
```

Comment:
We can do that but we lose some benefit of having this component which is to strictly enforce our form patterns. Breaking our form UX pattern should be a conscious decision that requires using the primitive button and configuring it or creating a new Form component. Those changes are stronger flags that we are creating a unique form than overriding props.

Reply:
I get your intention, but in this case we should omit the props that are assigned by this component so they can't be overwritten. Otherwise we'll end up with a broken implementation. Take this example:
```tsx
<FormikSubmitButton disabled={true} />
```

Here I'd expect the button to always be disabled, however that won't be the case since although it accepts a `disabled` prop it gets overwritten within the component itself.